src/movies.js

```diff
@@ -1,31 +1,141 @@
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(ListOfMovies) {
+  let ListOfDir = ListOfMovies.map(function (element) {
+    return element['director'];
+  });
+
+  return ListOfDir;
+}
+// console.log('---------');
+// console.log('---------');
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(Datalist) {
+  // we are going to find out how amny drama movies spielberg made
+  // we need to sort out all the movies speilberg made / map
+  // than we need to filter wich one of them are drama / filter
+  let listOfSpielberg = Datalist.filter(function (element) {
+    return element['director'] === 'Steven Spielberg';
+    // This is an element
+
+    // {
+    //   title: 'Indiana Jones and the Last Crusade',
+    //   year: 1989,
+    //   director: 'Steven Spielberg',
+    //   duration: '2h 7min',
+    //   genre: ['Action', 'Adventure', 'Fantasy'],
+    //   score: 8.3
+    // },
+  });
+  let listOfDrama = listOfSpielberg.filter(function (element) {
+    for (let i = 0; i < element['genre'].length; i++) {
+      let current_genre = element['genre'][i];
+      //return current_genre === "Drama";
+      if (current_genre === 'Drama') {
+        return true;
+      }
+    }
+    return false;
+  });
+  return listOfDrama.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(listOfmovies) {
+  if (listOfmovies.length === 0) {
+    return 0;
+  }
+  // I want all the scores first
+  let listOfScores = listOfmovies.map(function (element) {
+    return element['score'];
+    // I want to conculatethe average score
+  });
+  let listOfFilterdScore = listOfScores.filter(function (number) {
+    return number; // checking if a number is 'true'
+    // where are filtering movies that don't have a score
+  });
+  let sumScores = listOfFilterdScore.reduce(function (sum, number) {
+    return (sum += number);
+  }, 0);
+  // I am having a total amount of moviescore and deviding that bye the amount of movies
+  return parseFloat((sumScores / listOfmovies.length).toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(listOfmovies) {
+  let listofmoviesWithGenre = listOfmovies.filter(function (element) {
+    return element['genre'];
+  });
+
+  // cheak if these movies have drama
+  let listOfDrama = listofmoviesWithGenre.filter(function (movie) {
+    // i need to look through the list of genre and cheak if it containts drama
+    for (i = 0; i < movie['genre'].length; i++) {
+      if (movie['genre'][i] === 'Drama') {
+        return true;
+      }
+    }
+    return false;
+  });
```

Comment:
Again, `filter()` already iterates through your array, so using another loop is not neccesary ;)

Firstly you filter to only movies who have genre property. 
```
  let listofmoviesWithGenre = listOfmovies.filter(function (element) {
    return element['genre'];
  });
```

And you can do exactly same to get only dramas, for example:

```
  let listofmoviesWithGenre = listOfmovies.filter(function (element) {
    return element['genre'].includes('Drama');
  });
```